### ISSUE TITLE
Pre-vulcanize results.html

### ISSUE THREAD

**Issue by benshayden** [catapult-project/catapult#3925]
Telemetry vulcanizes benchmark results into results.html files using [results_renderer.py](https://github.com/catapult-project/catapult/blob/master/tracing/tracing/results_renderer.py). This module

1. dynamically uses py_vulcanize to collect all of the html source files from a catapult working directory checkout,
1. vulcanizes the html source files into a single large string, and then
1. appends `<histogram-json>` tags containing Histograms and shared Diagnostics.

A full catapult checkout is required in order to produce results.html files.

Pinpoint needs to produce results.html files from within the dashboard backend, which, IIUC, does not contain a full catapult checkout on disk in a format that py_vulcanize can support. (Please correct this if it's wrong!)
In order for Pinpoint to be able to produce results.html files, the Dashboard backend needs to contain a single HTML file that results_renderer.py can use as a template.

Should this vulcanized template file be vulcanized dynamically in the Dashboard deploy script, or by a separate script and statically checked into catapult?
Either way, the deploy script would need to set [`application_readable`](https://stackoverflow.com/questions/2630205/read-a-file-on-app-engine-with-python) so that results_renderer can read it.
Instead of using py_vulcanize dynamically, then, results_renderer would only need to read `results2_template_vulcanized.html` and append the `<histogram-json>` tags.

@dave-2 @simonhatch

**Comment by benshayden** [comment]
quest_generator needs to add --output-format=histograms
models/quest/read_value needs to read histogram dicts from histograms.json and store them in a member field.
models/job.py _Complete needs to get histogram dicts from all attempts' _last_execution.histogram_dicts and call RenderHistogramViewer with a StringStream or something and upload it to cloud storage.
Testing this is blocked on the Easy button on the job page #3883 .

How do we upload to cloud_storage from appengine?

Alternative: dashboard/histogram_viewer with multiple backends: dashboard database Histogram entities; open histograms.json file; pinpoint database Execution entities. This viewer might not require setting Polymer.useNativeShadow on other dashboard entities.

**Comment by dave-2** [comment]
Currently thinking that the alternative is the way to go - the histogram_viewer service.

It's awkward to plumb histogram data through from the execution to the Job, because it requires the Job to be aware of the implementation details of the `ReadValue` Execution. So it's easier if we can just call `job.AsDict()`, which makes the Execution handle its own serialization. Then we can pass the serialized Job JSON to the histogram viewer service, which can handle downloading and rendering the histograms.

The histograms are also going to be too big to store in a Job entity, so maybe we want to store the URLs or isolate hashes in the Executions instead of the full data.

**Comment by benshayden** [comment]
Opening a separate issue about the viewer service soon.
